server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/Routes.kt

```diff
@@ -0,0 +1,83 @@
+package com.xebia.functional.xef.server.http.routes
+
+import com.aallam.openai.api.BetaOpenAI
+import com.aallam.openai.api.chat.ChatCompletionRequest
+import com.aallam.openai.api.chat.ChatRole
+import com.xebia.functional.xef.auto.CoreAIScope
+import com.xebia.functional.xef.auto.PromptConfiguration
+import com.xebia.functional.xef.auto.llm.openai.*
+import com.xebia.functional.xef.auto.llm.openai.OpenAI.Companion.DEFAULT_CHAT
+import com.xebia.functional.xef.llm.Chat
+import com.xebia.functional.xef.llm.models.chat.Message
+import com.xebia.functional.xef.llm.models.chat.Role
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.util.pipeline.*
+import com.xebia.functional.xef.llm.models.chat.ChatCompletionRequest as XefChatCompletionRequest
+
+@OptIn(BetaOpenAI::class)
+fun Routing.routes() {
+    authenticate("auth-bearer") {
+        post("/chat/completions") {
+            val model: Chat = call.request.headers["xef-model"]?.let {
+                it.toOpenAIModel()
+            } ?: DEFAULT_CHAT
+            val token = call.principal<UserIdPrincipal>()?.name ?: throw IllegalArgumentException("No token found")
+            val scope = CoreAIScope(OpenAIEmbeddings(OpenAI(token).GPT_3_5_TURBO_16K))
+            val data = call.receive<ChatCompletionRequest>().toCore()
+            response<String, Throwable> {
+                model.promptMessage(
```

Comment:
Remember, when we are using `model.promptMessage` instead of `ai {}` block, we don't have conversations. 

I think that now is our best option, but we have to keep it in mind in case we want to have conversations in the future

Reply:
It automatically adds the `conversationId` if you pass it through as argument or is one of the methods that takes the VectorStore

Reply:
As this is based on the OpenAI API specification, it doesn't have a `conversationId`, instead they have the `user` that is making the call. And it's a `String`, so if you want we can use it as the `conversationId`. But I'm not sure.

Reply:
We can address it in a different PR, the conversation ID has to be persisted and looked up on each call somehow optionally